src/mrack/providers/podman.py

```diff
@@ -92,17 +122,46 @@ async def wait_till_provisioned(self, resource):
 
         if datetime.now() >= timeout_time:
             logger.warning(
-                f"{cont_id} was not provisioned within a timeout of" f" {timeout} mins"
+                f"{self.dsp_name}: {cont_id} was not provisioned within a timeout of"
+                f" {timeout} mins"
             )
         else:
-            logger.info(f"{cont_id} was provisioned in {prov_duration:.1f}s")
+            logger.info(
+                f"{self.dsp_name}: {cont_id} was provisioned in {prov_duration:.1f}s"
+            )
+
+        logger.debug(f"{self.dsp_name}: Resource: {object2json(server)}")
+
+        with open(os.path.expanduser(self.ssh_key), "r") as key_file:
+            key_content = key_file.read()
+
+        if not await self.podman.exec_command(cont_id, "mkdir -p /root/.ssh/"):
+            raise ProvisioningError(f"Could not copy public key to container {cont_id}")
+
+        if not await self.podman.exec_command(
+            cont_id, f'echo "{key_content}" >> /root/.ssh/authorized_keys'
+        ):
+            raise ProvisioningError(f"Could not copy public key to container {cont_id}")
+
+        if not await self.podman.exec_command(cont_id, "systemctl restart sshd"):
+            raise ProvisioningError(
+                f"Failed restarting sshd service in container {cont_id}"
+            )
 
-        logger.debug(object2json(server))
         return server
 
     async def delete_host(self, host_id):
         """Delete provisioned host."""
-        deleted = await self.podman.rm(host_id, force=True)
+        # first we inspect the container to find its networks
+        insp_data = await self.podman.inspect(host_id)
+        networks = insp_data[0]["NetworkSettings"]["Networks"]
```

Comment:
Are all networks take into account here? That means this deletes all networks, let's say one is using mrack and has a previous network in their machine, will this destroy that network as well?

Reply:
Nice, got it.